File: Leet Code/1. Two Sum.c
#include <stdio.h>
#include <stdlib.h>

int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int* result = (int*)malloc(2 * sizeof(int));
    *returnSize = 2;

    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
    }
    return NULL;
}

int main() {
    int nums[] = {2, 7, 11, 15};
    int target = 9;
    int returnSize;

    int* res = twoSum(nums, 4, target, &returnSize);

    printf("[%d, %d]\n", res[0], res[1]);
    free(res);

    system("pause");

    return 0;
}

File: Leet Code/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict1 = {}

        n = len(nums)

        for i in range(n):

            check = target - nums[i]
            
            if check in dict1:
                return [i,dict1[check]]
            
            dict1[nums[i]] = i

        return None